BucketHandlerService/Controllers/SolverBucketController.cs

```diff
@@ -0,0 +1,130 @@
+        //This means that this endpoint needs to be called before downloading the data everytime
+        responseData.ConfigURL = urlSigner.Sign(solverBucket, $"{solverName}.msc", TimeSpan.FromDays(1), HttpMethod.Get);
+        responseData.SolverURL = urlSigner.Sign(solverBucket, $"{solverName}", TimeSpan.FromDays(1), HttpMethod.Get);
+
+        var jsonData = JsonSerializer.Serialize(responseData).Replace("\\u0026", "&");
+        return Ok(jsonData);
+    }
+
+    [HttpPost]
+    public ActionResult PostSolver([FromBody] SolverRequest request)
+    {
+        if (client.ListObjects(solverBucket).Any(x => x.Name.StartsWith($"{request.SolverName}")))
+        {
+            return Conflict($"A solver with the name '{request.SolverName}' is already present");
+        }
+
+        var configContents = new ConfigResponse()
+        {
+            ID = $"id.{request.SolverName}",
+            Name = request.SolverName,
+            Executable = $"/Downloads/{request.SolverName}",
+            Version = request.Version,
+            SupportsMzn = request.SupportsMzn.ToUpper() == "TRUE",
+            SupportsFzn = request.SupportsFzn.ToUpper() == "TRUE"
+        };
+
+        var responseData = new SolverBucketResponse()
+        {
+            SolverName = request.SolverName,
+            ConfigData = JsonSerializer.Serialize(configContents),
+            MethodAllowed = "PUT"
+        };
+
+        var contentHeaders = new Dictionary<string, IEnumerable<string>>
+            {
+                { "Content-Type", new[] { "application/json" } }
+            };
+
+        UrlSigner.Options options = UrlSigner.Options.FromDuration(TimeSpan.FromDays(1));
+
+        UrlSigner.RequestTemplate solverTemplate = UrlSigner.RequestTemplate
+            .FromBucket(solverBucket)
+            .WithObjectName(request.SolverName)
+            .WithHttpMethod(HttpMethod.Put)
+            .WithContentHeaders(contentHeaders);
+
+        UrlSigner.RequestTemplate configTemplate = UrlSigner.RequestTemplate
+            .FromBucket(solverBucket)
+            .WithObjectName($"{request.SolverName}.msc")
+            .WithHttpMethod(HttpMethod.Put)
+            .WithContentHeaders(contentHeaders);
+
+        responseData.SolverURL = urlSigner.Sign(solverTemplate, options);
+        responseData.ConfigURL = urlSigner.Sign(configTemplate, options);
+
+        var jsonData = JsonSerializer.Serialize(responseData).Replace("\\u0026", "&").Replace("\\u0022", "\"");
+        return Ok(jsonData);
+    }
+
+    private UrlSigner InitializeUrlSigner()
+    {
+        UrlSigner signer;
+        using (var stream = GetSecretCredentials())
+        {
+            signer = UrlSigner.FromServiceAccountData(stream);
+        }
+        return signer;
+    }
+
+    private Stream GetSecretCredentials()
+    {
+        var secretResourceName = "projects/859134286483/secrets/default_service_account_key/versions/1";
```

Comment:
Dont no if there is time for it at this point, but maybe have this in env instead.